nixos/modules/services/hardware/display.nix

```diff
@@ -0,0 +1,254 @@
+          outputs."DP-1".edid = "PG278Q_60.bin";
+          outputs."DP-1".mode = "e";
+        }
+      '';
+    };
+  };
+
+  config = lib.mkMerge [
+    {
+      hardware.display.edid.packages =
+        lib.optional (cfg.edid.modelines != null) cfg.edid.modelines
+        ++ lib.optional (cfg.edid.linuxhw != null) cfg.edid.linuxhw;
+
+      boot.kernelParams =
+        # forcing video modes
+        lib.trivial.pipe cfg.outputs [
+          (lib.attrsets.filterAttrs (_: spec: spec.mode != null))
+          (lib.mapAttrsToList (output: spec: "video=${output}:${spec.mode}"))
+        ]
+        ++
+        # selecting EDID for displays
+        lib.trivial.pipe cfg.outputs [
+          (lib.attrsets.filterAttrs (_: spec: cfg.edid.applyWithKernelParameters && spec.edid != null))
+          (lib.mapAttrsToList (output: spec: "${output}:edid/${spec.edid}"))
+          (builtins.concatStringsSep ",")
+          (p: lib.optional (p != "") "drm.edid_firmware=${p}")
+        ]
+      ;
+    }
+    (
+      let
+        edids = lib.attrsets.filterAttrs (connector: output: cfg.edid.applyAtRuntime && output.edid != null) cfg.outputs;
+        loads = lib.pipe edids [
+          (lib.attrsets.mapAttrsToList (connector: output: "load ${connector} ${output.edid}"))
+          (builtins.concatStringsSep "\n")
+        ];
+      in
+      lib.mkIf (edids != { }) {
+        systemd.services.display-edid-apply = {
+          before = [ "graphical.target" "multi-user.target" ];
+          wantedBy = [ "graphical.target" "multi-user.target" ];
+          enable = true;
+          serviceConfig.Type = "oneshot";
+          serviceConfig.RemainAfterExit = true;
+          script = ''
+            export PATH="${lib.makeBinPath (with pkgs; [ findutils coreutils ])}:$PATH"
+
+            edid_paths=(
+              /run/current-system/firmware/edid/
+              ${lib.optionalString (cfg.edid.modelines != null) "${cfg.edid.modelines}/lib/firmware/edid/"}
+              ${lib.optionalString (cfg.edid.linuxhw != null) "${cfg.edid.linuxhw}/lib/firmware/edid/"}
+            )
+
+            find_connector() {
+              mapfile -t candidates < <(find /sys/kernel/debug/dri -maxdepth 2 -name "$1")
+              echo -n "''${candidates[0]}"
+            }
+
+            find_edid() {
+              mapfile -t candidates < <(find "''${edid_paths[@]}" -maxdepth 1 -name "$1")
+              echo -n "''${candidates[0]}"
+            }
+
+            load() {
+              local connector="$1" filename="$2" connectorPath
+              connectorPath="$(find_connector "$1")"
+              edidPath="$(find_edid "$2")"
+
+              printf "Applying '%s' to '%s' output" "$edidPath" "$connectorPath" >&2
+              cat "$edidPath" >"$connectorPath/edid_override"
+              echo 1 >"$connectorPath/trigger_hotplug"
+            }
+
+            ${loads}
+          '';
+        };
+      }
+    )
+    (lib.mkIf (cfg.edid.packages != null) {
```

Comment:
We should probably hide this behind en enable flag. Some modules, eg in nixos-hardware might fill this and having to use mkForce to clear it, is not that nice.